### DISCUSSION TITLE
Gradle-Plugin: Are Tasks with TypeResolution not using the same compiler options as configured for the project?

### DISCUSSION THREAD

**Discussion by mervyn-mccreight** [detekt/detekt#3629] [General]
Hi folks :-)

When using tasks like e.g. `detektMain` I figured out that it displays a lot of compiler warnings.

E.g. I have opted-in kotlin.time by passing a compiler-flag like the following in my `build.gradle.kts`
```kotlin
tasks.withType<KotlinCompile> {
    kotlinOptions.freeCompilerArgs += "-Xuse-experimental=kotlin.time.ExperimentalTime"
}
```
so I do not have to opt-in it all over the source-code.

`detektMain` does not seem to pick this up and is constantly complaining about it with messages like this:

```
<path/to/source/File.kt>: error: this declaration is experimental and its usage must be marked with '@kotlin.time.ExperimentalTime' or '@OptIn(kotlin.time.ExperimentalTime::class)
```

which does not look very nice. Is the validation even working for these files?
The task itself "passes" if no source-file violates any rules, but in my perception it is misleading that the terminal is full of red text containing the term "error" if everything is fine.

Is it intended that detekt is not using the same compiler flag as the project uses to check against rules using type-resolution?
Or am I missing some configuration so that detekt does this?

**Comment by BraisGabin** [comment]
Good point. It's a pity that we can't move a discussion to an issue because I think that this is an issue. I'm not sure if we could do something about it but we could, at least, write something in our documentation to how to solve this problem.

Probably this issue is gone with https://github.com/detekt/detekt-compiler-plugin

**Reply by mervyn-mccreight** [reply]
I'll open an issue for this then :)

**Reply by cortinico** [reply]
> Probably this issue is gone with detekt/detekt-compiler-plugin

Agree. I believe that this, and other compilation related tweaks, should not be tackled and we should push to productionize the compiler plugin.

**Comment by schalkms** [comment]
>  It's a pity that we can't move a discussion to an issue because I think that this is an issue.

It's possible to create a new issue and delete the discussion afterwards.
